test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -172,15 +172,20 @@ create_configs()
     echo "salvage=0" >> $file_name                  # Faster runs
     echo "stress.checkpoint=0" >> $file_name        # Faster runs
     echo "timer=4" >> $file_name
-    echo "verify=1" >> $file_name                   # Faster runs
+    echo "verify=1" >> $file_name
+    # WT-8601 has not been backported to 4.2 and transactions are expected to be timestamped.
+    if [ "$branch_name" == "mongodb-4.2" ] ; then
+        echo "transaction.timestamps=1" >> $file_name # WT-7545 - Older releases can't do non-timestamp transactions
+    else
+        echo "transaction.timestamps=0" >> $file_name # WT-8601 - Timestamps do not work with logged tables
```

Comment:
Does this change remove the non-timestamp coverage in the compatibility test? I guess line 169 (enabled logging)  requires this setting here.

Reply:
Since WT-8601, 6.0+ versions were not tested with timestamps as it was incompatible. The other versions would have timestamps enabled.
Now that we have backported WT-8601 to 4.4 and 5.0, they are losing the timestamps coverage.
The only one that keeps the timestamp coverage is 4.2.

One way to mitigate this is to randomise `logging` and `transaction.timestamps`. We could enforce those two only for 4.2 and let test/format randomise the values for the other versions. If an incompatible combination is created, test/format handles it by leaving `logging` the priority.

Let me know what you think.

Reply:
> One way to mitigate this is to randomise logging and transaction.timestamps. We could enforce those two only for 4.2 and let test/format randomise the values for the other versions. If an incompatible combination is created, test/format handles it by leaving logging the priority.

That sounds much better to me! 
Though not totally sure why logging was enabled initially, I guess some testing cycles will reveal whether it's a must.

Reply:
I have tried the suggested changes [here](https://spruce.mongodb.com/version/6492e780562343859d14dad9/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC). However we are running into another issue, I have created WT-11213 for this. I am not sure yet if they should be fixed together.

Reply:
To avoid this issue, we may want to leave the randomisation (hence the combination `logging=0` and `txn.timestamps=1`) until WT-11213 is solved.

Reply:
sounds good to me.

Reply:
@lukech, if you are ok with this, we are gonna stick to those changes and have a follow-up ticket after WT-11213 is solved to be able to randomise `logging` and `txn.timestamps`.

Reply:
After some testing, the issue is still there. I understand better the script and we are trying to make sure to generate a config that is compatible with the branch that has the most restrictions.
We are in the following situation:
- We cannot have logging and txn.timestamps enabled at the same time for all the branches except 4.2
- 4.2 needs to have txn.timestamps enabled
- 4.2 is used to run against a config generated by a different branch and **this is why it is failing**. All the branches are generating logging=1 and txn.timestamps=0 but 4.2 needs txn.timestamps=1 to work.

Reply:
Thanks @etienneptl. I'm okay with the plan to use a follow-on ticket to investigate/implement `logging` randomization for newer branches.